extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JakartaEnablement.java

```diff
@@ -57,18 +56,25 @@ private static class JakartaTransformer {
 
         private final Logger logger;
         private final ActionContext ctx;
+        private static final Map<String, String> renames = Collections.singletonMap(forbiddenName(), "jakarta.transaction");
 
         JakartaTransformer() {
             logger = LoggerFactory.getLogger("JakartaTransformer");
-            Map<String, String> renames = new HashMap<>();
-            //N.B. we enable only this transformation, not the full set of capabilities of Eclipse Transformer;
+            //N.B. we enable only this single transformation of package renames, not the full set of capabilities of Eclipse Transformer;
             //this might need tailoring if the same idea gets applied to a different context.
-            renames.put("javax.transaction", "jakarta.transaction");
             ctx = new ActionContextImpl(logger,
                     new SelectionRuleImpl(logger, Collections.emptyMap(), Collections.emptyMap()),
                     new SignatureRuleImpl(logger, renames, null, null, null, null, null, Collections.emptyMap()));
         }
 
+        //Need to prevent the Eclipse Transformer - which is run on this whole code base - to actually replace this name:
+        private static String forbiddenName() {
+            StringBuilder sb = new StringBuilder("java")
+                    .append("x.")
+                    .append("transaction");
```

Comment:
IIRC, I just fixed it by doing a `"javax." + "transaction"` where I faced the issue.

Reply:
But given you're transforming the bytecode and not the source, maybe it wouldn't work.

Reply:
yea, pretty sure that would have worked too. But didn't want to take chances - spent enough on this :)